SwiftReflector/OverrideBuilder.cs

```diff
@@ -221,10 +255,18 @@ void CopyVirtualMethods ()
 			else
 				HandleSuperClassVirtualMethods (OriginalClass);
 			IndexOfFirstNewVirtualMethod = OverriddenVirtualMethods.Count;
-			OverriddenVirtualMethods.AddRange (VirtualMethodsForClass (OriginalClass).Select (m => MarkOverrideSurrogate (m, Reparent (new FunctionDeclaration (m), OverriddenClass))));
-			var members = isProtocol ? EveryProtocolExtension.Members : OverriddenClass.Members;
+			if (isProtocol && hasAssociatedTypes) {
+				OverriddenVirtualMethods.AddRange (VirtualMethodsForClass (OriginalClass).Select (m => MarkOverrideSurrogate (m, Reparent (RebuildFunctionDeclarationWithAssociatedTypes (OverriddenClass, m), OverriddenClass))));
+			} else {
+				OverriddenVirtualMethods.AddRange (VirtualMethodsForClass (OriginalClass).Select (m => MarkOverrideSurrogate (m, Reparent (new FunctionDeclaration (m), OverriddenClass))));
+			}
```

Comment:
This is personal, but for debugging, I would create a range var, add the result of the LINQ and then do a single call to AddRange.

```csharp
IEnumberable range;
if (isProtocol && hasAssociatedTypes) {
 				range = VirtualMethodsForClass (OriginalClass).Select (m => MarkOverrideSurrogate (m, Reparent (RebuildFunctionDeclarationWithAssociatedTypes (OverriddenClass, m), OverriddenClass))));
 			} else {
 				range = VirtualMethodsForClass (OriginalClass).Select (m => MarkOverrideSurrogate (m, Reparent (new FunctionDeclaration (m), OverriddenClass))));
 			}
OverriddenVirtualMethods.AddRange (range);
```csharp
When we debug, we will be able to see the range value, will be simpler.